src/persist/src/operators/replay.rs

```diff
@@ -75,6 +79,10 @@ where
                                     debug_assert!(snapshot_since.less_equal(ts));
                                 }
                                 session.give(x);
+                                if idx + 1 >= outputs_per_yield {
```

Comment:
it was not obvious to me that `enumerate` will restart counting from zero every time we re-enter the operator - maybe worth a comment?

Reply:
Done